src/browser/api/application.js

```diff
@@ -837,6 +838,15 @@ Application.emitRunRequested = function(identity, userAppConfigArgs) {
     }
 };
 
+Application.reloadPreloadScripts = function(identity, preloadScripts, callback) {
+    (preloadScripts || []).map(preload => {
+        electronApp.vlog(1, `clear preload script ${getIdentifier(preload)}`);
+        deletePreloadScriptState(getIdentifier(preload));
+    });
+    System.downloadPreloadScripts(identity, preloadScripts, callback);
+
+};
+
```

Comment:
Perhaps we should consider checking that all the requested "scripts to re-download" were previously supplied as window options. As-written, the API allows download of arbitrary files to the cache, even if they were never forward-declared. Such a feature may / may not be desirable.

Suggest checking that all requested scripts here already have an entry in the preload script cache.

Reply:
if a script is not listed in window options, it will not be applied during reload